drivers/led/sn32/rgb_matrix_sn32f24xb.c

```diff
@@ -0,0 +1,426 @@
+#include "rgb_matrix.h"
+#include "sn32f24xb.h"
+#include "debounce.h"
+
+#if !defined(RGB_MATRIX_HUE_STEP)
+#    define RGB_MATRIX_HUE_STEP 8
+#endif
+
+#if !defined(RGB_MATRIX_SAT_STEP)
+#    define RGB_MATRIX_SAT_STEP 16
+#endif
+
+#if !defined(RGB_MATRIX_VAL_STEP)
+#    define RGB_MATRIX_VAL_STEP 16
+#endif
+
+#if !defined(RGB_MATRIX_SPD_STEP)
+#    define RGB_MATRIX_SPD_STEP 16
+#endif
+
+#ifdef MATRIX_NO_SCAN
+#   define ROWS_PER_HAND (MATRIX_ROWS)
+
+#   ifndef MATRIX_IO_DELAY
+#       define MATRIX_IO_DELAY 30
+#   endif
+#endif
+/*
+    COLS key / led
+    SS8050 transistors NPN driven low
+    base      - GPIO
+    collector - LED Col pins
+    emitter   - VDD
+
+    VDD     GPIO
+    (E)     (B)
+     |  PNP  |
+     |_______|
+         |
+         |
+        (C)
+        LED
+
+    ROWS RGB
+    SS8550 transistors PNP driven high
+    base      - GPIO
+    collector - LED RGB row pins
+    emitter   - GND
+
+        LED
+        (C)
+         |
+         |
+      _______
+     |  NPN  |
+     |       |
+    (B)     (E)
+    GPIO    GND
+*/
+static uint8_t chan_col_order[LED_MATRIX_COLS] = {0}; // track the channel col order
+static uint8_t current_row = 0; // LED row scan counter
+static uint8_t row_idx = 0; // key row scan counter
+#ifdef MATRIX_NO_SCAN
+extern matrix_row_t raw_matrix[MATRIX_ROWS]; //raw values
+extern matrix_row_t matrix[MATRIX_ROWS]; //debounced values
+matrix_row_t current_matrix[MATRIX_ROWS] = {0}; //scan values
+bool has_changed = false; // matrix update check
+#endif
+static const uint32_t periodticks = 256;
+static const uint32_t freq = (RGB_MATRIX_HUE_STEP * RGB_MATRIX_SAT_STEP * RGB_MATRIX_VAL_STEP * RGB_MATRIX_SPD_STEP * RGB_MATRIX_LED_PROCESS_LIMIT);
+static const pin_t led_row_pins[LED_MATRIX_ROWS_HW] = LED_MATRIX_ROW_PINS; // We expect a R,B,G order here
+static const pin_t led_col_pins[LED_MATRIX_COLS] = LED_MATRIX_COL_PINS;
+RGB led_state[RGB_MATRIX_LED_COUNT]; // led state buffer
+bool enable_pwm = false;
```

Comment:
```suggestion
static RGB led_state[RGB_MATRIX_LED_COUNT]; // led state buffer
static bool enable_pwm = false;
```
(unless there is some external usage that I don't see for some reason, but then these variable names are not really good).